### PR TITLE
Invoking Context Menu from ListRow component

### PR DESCRIPTION
**Pull request by tidy-dev**
Based on https://github.com/desktop/desktop/pull/16108

(This diff looks funny because I have development merged into this branch and that pull request above does not.)

## Description
This PR refactor the callback of `onContextMenu` to be on the focus div element of each row of the changes list. That is the div in the `<ListRow>` component. The reason being is that the keyboard shortcuts (i.e. Shift + F10 on Windows and VO "VoiceOver" + Shift + M on MacOS) to invoke a context menu will only do so for the focused div which for a selectable list item is the one with the role of `option`

## Release notes
Notes: [Improved] The context menu for a file in changed files list can be invoked by keyboard shortcuts.

### CODE REVIEW

**Comment by tidy-dev** [comment]
After much trial by error.. I stumbled upon the fact that @niik changes in https://github.com/desktop/desktop/pull/16108 solved the problem of the incorrect row context menu being rendered on macOS when I went to start simplying the Grid rendering to see what thing could be impacting it and I applied some of @niik's changes and it just worked! I can't really explain why.. I am assuming something about roles where impacting this. :/

File: app/src/ui/lib/list/list-row.tsx [hunk 1]
```diff
@@ -94,6 +101,12 @@ export class ListRow extends React.Component<IListRowProps, {}> {
     this.props.onRowBlur?.(this.props.rowIndex, e)
   }
 
+  private onContextMenu = (e: React.MouseEvent<HTMLDivElement>) => {
+    console.log(this.props.rowIndex)
+    console.log(e.currentTarget)
```
Comment (niik) [inline app/src/ui/lib/list/list-row.tsx]: Can we get rid of these please?
```suggestion
```

**Review by niik — CHANGES_REQUESTED** [review]
One tiny request

File: app/src/ui/lib/list/list-row.tsx (hunk 1 above)
Comment (tidy-dev) [reply app/src/ui/lib/list/list-row.tsx]: Woops.. Absolutely!

**Review by niik — APPROVED** [review]

**Pull request merged**